org.omg.kerml.xpect.tests/library/SequenceFunctions.kerml

```diff
@@ -0,0 +1,47 @@
+package SequenceFunctions {
+	private import Base::Anything;
+	private import ScalarValues::*;
+	private import ControlFunctions::exists;
+	private import ControlFunctions::select;
+	private import ControlFunctions::reject;
+
```

Comment:
All the `seq` parameters on the SequenceFunctions should actually be marked as `ordered nonunique`. Indeed, many of the functions really rely on this semantically -- e.g., indexing only makes sense for `ordered` and computing the size has to count duplicate elements, if there are any. So, these parameters really should be sequences. And, while this may seem like the most restrictive case, it is actually the most general, because any unordered collection can be treated as just having an arbitrary order and a collection with unique values is just a special case of one that is non-unique. (This is also the approach taken for SequenceFunctions in Alf.) I have pushed an update to SequenceFunctions that corrects the declaration of the `seq` parameters.

In KerML, as in UML/SysML v1, the basic construct for modeling "collections" of things is multiplicity. The Collection types operated on by the CollectionFunctions are _defined_ using multiplicity on their `elements` feature. So the CollectionFunctions have to be defined in terms of the `SequenceFunctions`, not the other way around.

Reply:
👍 agreed and understood